azure-quantum/azure/quantum/target/microsoft/elements/dft/job.py

```diff
@@ -0,0 +1,56 @@
+import json
+from typing import Any, Dict
+from azure.quantum.job.job import Job, DEFAULT_TIMEOUT
+from azure.quantum._client.models import JobDetails
+
+class MicrosoftElementsDftJob(Job):
+    """
+    A dedicated job class for jobs from the microsoft.dft target.
+    """
+
+    def __init__(self, workspace, job_details: JobDetails, **kwargs):
+        super().__init__(workspace, job_details, **kwargs)
+
+    def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT) -> Dict[str, Any]:
+        try:
+            job_results = super().get_results(timeout_secs)
+            return job_results["results"]
+        except RuntimeError:
+            if self.details.status == "Failed":
+                job_blob_properties = self.download_blob_properties(self.details.output_data_uri)
+                if job_blob_properties.size > 0:
+                    job_failure_data = self.download_data(self.details.output_data_uri)
+                    raise DftRuntimeError("An error occurred during DFT-job execution. " \
+                                          "Use error.get_details() for more information.", job_failure_data)
+            raise
+
+
+class DftRuntimeError(RuntimeError):
```

Comment:
Consider having a base class like JobError so we can establish a pattern across targets.

Reply:
or maybe FailedJobWithResultError since we seem to prefer Result over Output

Reply:
Renamed the error-class to `JobFailedWithResultsError` as the method producing this error is called `get_results()` I thought making "results" plural would align.